fineract-loan/src/main/java/org/apache/fineract/portfolio/interestpauses/service/InterestPauseReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @Override
+    @Transactional
+    public Long createInterestPauseByLoanId(Long loanId, String startDate, String endDate, String dateFormat, String locale) {
+        this.context.authenticatedUser();
+
+        final Loan loan = loanRepository.findById(loanId)
+                .orElseThrow(() -> new IllegalArgumentException("Loan not found with ID: " + loanId));
+
+        validateInterestPauseDates(loan, startDate, endDate, dateFormat, locale);
+
+        LocalDate startLocalDate = parseAndValidateDate(startDate, dateFormat, locale);
+        LocalDate endLocalDate = parseAndValidateDate(endDate, dateFormat, locale);
+
+        LoanTermVariations variation = new LoanTermVariations(LoanTermVariationType.INTEREST_PAUSE.getValue(), startLocalDate, null,
+                endLocalDate, false, loan);
+
+        LoanTermVariations savedVariation = loanTermVariationsRepository.save(variation);
+        loanTermVariationsRepository.flush();
+
+        return savedVariation.getId();
+    }
+
+    private void validateInterestPauseDates(Loan loan, String startDateStr, String endDateStr, String dateFormat, String locale) {
+        if (startDateStr == null || endDateStr == null) {
+            throw new IllegalArgumentException("Both `startDate` and `endDate` parameters are mandatory.");
+        }
+
+        LocalDate startDate = parseAndValidateDate(startDateStr, dateFormat, locale);
+        LocalDate endDate = parseAndValidateDate(endDateStr, dateFormat, locale);
+
+        if (startDate.isBefore(loan.getSubmittedOnDate())) {
+            throw new IllegalArgumentException("Interest pause start date cannot be earlier than loan start date.");
+        }
+
+        if (endDate.isAfter(loan.getMaturityDate())) {
+            throw new IllegalArgumentException("Interest pause end date cannot be later than loan maturity date.");
+        }
+
+        if (!endDate.isAfter(startDate)) {
+            throw new IllegalArgumentException("Interest pause end date must be later than the interest pause start date.");
+        }
+    }
+
+    private LocalDate parseAndValidateDate(String date, String dateFormat, String locale) {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateFormat).withLocale(new Locale(locale));
+            return LocalDate.parse(date, formatter);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException(
+                    "The parameter `date` is invalid based on the dateFormat: `" + dateFormat + "` and locale: `" + locale + "` provided.",
+                    e);
+        }
+    }
+
+    private List<InterestPauseData> mapToInterestPauseData(List<LoanTermVariationsData> variations) {
+        return variations.stream()
+                .map(variation -> new InterestPauseData(variation.getId(), variation.getTermVariationApplicableFrom().toString(),
+                        variation.getDateValue().toString(), DateUtils.DEFAULT_DATE_FORMAT, "en"))
```

Comment:
Hardcoded locale is incorrect. We might not needed at all. Return the system locale. If we change the response field to return `LocalDate` we can use the date formatter from the converter: `LocalDateJsonConverter`